services/scheduler/scheduler.go

```diff
@@ -0,0 +1,330 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package scheduler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/percona/pmm-managed/models"
+
+	"github.com/AlekSi/pointer"
+	"github.com/go-co-op/gocron"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+)
+
+// Service is responsive for executing tasks and storing them to DB.
+type Service struct {
+	db        *reform.DB
+	scheduler *gocron.Scheduler
+	l         *logrus.Entry
+	tasks     map[string]context.CancelFunc
+	taskMx    sync.RWMutex
+	jobs      map[string]*gocron.Job
+	jobsMx    sync.RWMutex
+	mx        sync.Mutex
```

Comment:
I would suggest to sort fields a different way: https://dmitri.shuralyov.com/idiomatic-go#mutex-hat
```suggestion
	db        *reform.DB
	l         *logrus.Entry
	
	mx        sync.Mutex
	scheduler *gocron.Scheduler

	taskMx    sync.RWMutex
	tasks     map[string]context.CancelFunc

	jobsMx    sync.RWMutex
	jobs      map[string]*gocron.Job
```